MH4U Dex/CoreDataController.m

```diff
@@ -0,0 +1,430 @@
+        NSString *itemName = areaDropDict[@"item"];
+        Item *item = (Item *)[self getUniqueEntityWithEntityName:@"Item" withNameAttribute:itemName];
+        if (!item) {
+            NSEntityDescription *newItemDescription = [NSEntityDescription entityForName:@"Item" inManagedObjectContext:self.managedObjectContext];
+            item = [[Item alloc] initWithEntity:newItemDescription insertIntoManagedObjectContext:self.managedObjectContext];
+            item.name = itemName;
+        }
+        [item.areaSourceSet addObject:drop];
+        drop.item = item;
+        
+        // The Drop should now exist, and be properly hooked up to the area, which exists and is hooked up to the region.
+    }
+}
+
+# pragma mark - Helper Methods
+
+- (NSDictionary *)loadDictionaryFromJsonFileNamed:(NSString *)fileName
+{
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:fileName ofType:@"json"];
+    NSError *error;
+    NSString *fileContents = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+    if (error) {
+        NSLog(@"Error reading file: %@", error.localizedDescription);
+        return nil;
+    }
+    return (NSDictionary *)[NSJSONSerialization JSONObjectWithData:[fileContents dataUsingEncoding:NSUTF8StringEncoding]
+                                                           options:0
+                                                             error:nil];
+}
+
+- (NSArray *)loadArrayFromJsonFileNamed:(NSString *)fileName
+{
+    NSString *filePath = [[NSBundle mainBundle] pathForResource:fileName ofType:@"json"];
+    NSError *error;
+    NSString *fileContents = [NSString stringWithContentsOfFile:filePath encoding:NSUTF8StringEncoding error:&error];
+    if (error) {
+        NSLog(@"Error reading file: %@", error.localizedDescription);
+        return nil;
+    }
+    return (NSArray *)[NSJSONSerialization JSONObjectWithData:[fileContents dataUsingEncoding:NSUTF8StringEncoding]
+                                                      options:0
+                                                        error:nil];
+}
+
+- (BOOL)didSuccessfullySaveContext
+{
+    NSError *error = nil;
+    if (![self.managedObjectContext save:&error]) {
+        NSLog(@"Unable to save managed object context.");
+        NSLog(@"%@, %@", error, error.localizedDescription);
+        return NO;
+    }
+    return YES;
+}
+
+- (AreaDrop *)getUniqueAreaDropWithDecimalID:(NSString *)decimalID
+{
+    NSError *error = nil;
+    NSFetchRequest *request = [[NSFetchRequest alloc] initWithEntityName:@"AreaDrop"];
+    NSPredicate *predicate = [NSPredicate predicateWithFormat:@"%K == %@", @"idDecimalString", decimalID];
+    [request setPredicate:predicate];
+    
+    NSArray *results = [self.managedObjectContext executeFetchRequest:request error:&error];
+    if (!error) {
+        if (results.count == 0) {
+            // Object not found.
+            return nil;
+        } else if (results.count == 1) {
```

Comment:
Since you returned out of the `if` block, you could just go into the next `if` statement without the chaining `else`.